x-pack/plugins/lens/public/heatmap_visualization/types.ts

```diff
@@ -32,7 +33,8 @@ export type HeatmapLayerState = SharedHeatmapLayerState & {
 };
 
 export type HeatmapVisualizationState = HeatmapLayerState & {
-  palette?: PaletteOutput;
+  // need to store the current accessor to reset the color stops at accessor change
+  palette?: PaletteOutput<CustomPaletteParams> & { accessor: string };
```

Comment:
Can this conflict with existing saved objects which do not have an `accessor` property? I couldn't break it but maybe I missed a configuration?

Reply:
That is required to trigger a reset on the palette configuration when the operation changes.
To reproduce a situation without that, try the following:
* Configure an operation with some custom color stops with percentage
* Now change the operation in the dimension panel => inconsistent state

I tried other approaches as well and this one was the cleanest I found so far.

Reply:
Tested a previously saved heatmap: it migrated to the new format, with color being remapped to the new 5 stops format. Couldn't break it either.